packages/react-testing/src/tests/e2e.test.tsx

```diff
@@ -173,7 +175,8 @@ describe('@shopify/react-testing', () => {
       );
     }
 
-    it('releases any stale promises when component is destroyed', async () => {
+    // eslint-disable-next-line no-process-env
+    itIf(process.env.REACT_VERSION !== '17')('releases any stale promises when component is destroyed', async () => {
```

Comment:
I discovered that React exports a `version` variable that contains the what the name suggests - https://unpkg.com/browse/react@17.0.2/cjs/react.development.js#L2331

Perhaps we could use that? Something like:

```js
const isReact18OrMore = parseInt(version.split('.')[0], 10) >= 18;

// ...

itIf(isReact18OrMore)(...
```